File: client/src/app/core/models/view-models/audit-log-view-model.ts
import { Page, PageQuery } from '../pagination';
import { EntityType } from '../entity-type';
import { ActivityType } from './activity-view-model';

export interface AuditLogViewModel {
  id: number;
  occurredAt: Date;
  userId: string;
  userDisplayName: string;
  userPictureUrl?: string;
  type: ActivityType;
  entityType: EntityType;
  entityId?: number;
  workspaceSlug?: string;
  projectSlug?: string;
  boardSlug?: string;
  meta?: Record<string, unknown>;
}

export type AuditLogPage = Page<AuditLogViewModel>;

export interface AuditActivityPoint {
  date: string;
  count: number;
}

export interface AuditLogFilter extends PageQuery {
  userId?: string;
  entityType?: EntityType;
  activityType?: ActivityType;
  from?: string;
  to?: string;
}

File: server/Netptune.Core/ViewModels/Audit/AuditLogViewModel.cs
using System.Text.Json;

using Netptune.Core.Enums;

namespace Netptune.Core.ViewModels.Audit;

public class AuditLogViewModel
{
    public int Id { get; set; }
    public DateTime OccurredAt { get; set; }
    public string UserId { get; set; } = null!;
    public string UserDisplayName { get; set; } = null!;
    public string? UserPictureUrl { get; set; }
    public ActivityType Type { get; set; }
    public EntityType EntityType { get; set; }
    public int? EntityId { get; set; }
    public string? WorkspaceSlug { get; set; }
    public string? ProjectSlug { get; set; }
    public string? BoardSlug { get; set; }
    public JsonDocument? Meta { get; set; }
}
